radixdlt-java/src/main/java/com/radixdlt/client/application/translate/data/SendMessageToParticleGroupsMapper.java

```diff
@@ -125,7 +125,11 @@ public SendMessageToParticleGroupsMapper(
 					.build();
 			particles.add(SpunParticle.up(encryptorParticle));
 
-			payload = sharedKey.getPublicKey().encrypt(action.getData());
+			try {
+				payload = sharedKey.getPublicKey().encrypt(action.getData());
+			} catch (ECIESException e) {
+				throw new IllegalStateException("Failed to encrypt data", e);
```

Comment:
Curiously, this exception for failing to encrypt something has a different message than the other exception about failing to encrypt something (which says "Expected to always be able to encrypt"). Why are the expectations different?

Reply:
No reason, this is what happens when using constructor taking a string :P can change to:

```java
throw new IllegalStateException("Expected to always be able to encrypt", e);
```

But harmonizing exception will be done in coming PR.